packages/permission-controller/src/PermissionController.ts

```diff
@@ -1388,12 +1385,11 @@ export class PermissionController<
             default: {
               // This type check ensures that the switch statement is
               // exhaustive.
-              const _exhaustiveCheck: never = mutatorResult;
+              const exhaustiveCheck: never = mutatorResult;
               throw new Error(
                 `Unrecognized mutation result: "${
-                  // TODO: Replace `any` with type
-                  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-                  (_exhaustiveCheck as any).operation
+                  // @ts-expect-error We need to override the "never"
+                  exhaustiveCheck.operation
                 }"`,
```

Comment:
The exhaustiveness checking here seems both incorrect and unnecessary:
- The switch expression is `mutatorResult.operation`, not `mutatorResult`
- The type of the switch expression is an enum, which supports exhaustiveness checking without requiring the [explicit `never` assignment trick from the docs](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#exhaustiveness-checking)  (note that `operation` evaluates to `never` in the code below).

We could simplify the default clause:

```diff
diff --git a/packages/permission-controller/src/PermissionController.ts b/packages/permission-controller/src/PermissionController.ts
index 1f724abd..17d41065 100644
--- a/packages/permission-controller/src/PermissionController.ts
+++ b/packages/permission-controller/src/PermissionController.ts
@@ -1350,7 +1350,8 @@ export class PermissionController<
           // The mutator may modify the caveat value in place, and must always
           // return a valid mutation result.
           const mutatorResult = mutator(targetCaveat.value);
-          switch (mutatorResult.operation) {
+          const { operation } = mutatorResult;
+          switch (operation) {
             case CaveatMutatorOperation.noop:
               break;
 
@@ -1383,15 +1384,9 @@ export class PermissionController<
               break;
 
             default: {
-              // This type check ensures that the switch statement is
-              // exhaustive.
-              const exhaustiveCheck: never = mutatorResult;
-              throw new Error(
-                `Unrecognized mutation result: "${
-                  // @ts-expect-error We need to override the "never"
-                  exhaustiveCheck.operation
-                }"`,
-              );
+              // Overriding as `never` is the expected result of exhaustiveness checking, and is intended to represent unchecked exception cases.
+              // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+              throw new Error(`Unrecognized mutation result: "${operation}"`);
             }
           }
         });
```